core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -178,6 +179,34 @@ inline void configure_shmem_preference(KernelFuncPtr const& func,
 #endif
 }
 
+template <class Policy>
+std::enable_if_t<Policy::experimental_contains_desired_occupancy>
+modify_launch_configuration_if_desired_occupancy_is_specified(
+    Policy const& policy, cudaDeviceProp const& properties,
+    cudaFuncAttributes const& attributes, dim3 const& block, int& shmem,
+    bool& prefer_shmem) {
+  int const block_size        = block.x * block.y * block.z;
+  int const desired_occupancy = policy.impl_get_desired_occupancy().value();
+
+  size_t const shmem_per_sm_prefer_l1 = get_shmem_per_sm_prefer_l1(properties);
+  size_t const static_shmem           = attributes.sharedSizeBytes;
+  int active_blocks = properties.maxThreadsPerMultiProcessor / block_size *
+                      desired_occupancy / 100;
```

Comment:
I think this is iffy. Say maxThreadsPerMultiProc is 2048, block size is 700, desired_occupancy 33 doesn't that get me a zero here? (2048/700 = 2.9.. = 2 * 33 = 66 / 100 = 0?) even though 700 /2048 > 0.33?
In either case don't we need to make active_blocks ==0 into active_blocks=1 before dividing in the next line?

Reply:
And even if you use 670 as the block size (which makes you go just below 0.33) you end up with zero. Since 3*33 is 99 and that still divides away to zero. This problem can be solved by just making the computation use double. We also probably should round and not round towards zero. Like if it computes that 1.9 blocks give you the desired occupancy, I think we should round to 2 instead of 1.

Reply:
In either case the potential divide by zero needs to be fixed.

Reply:
Fixed